CalculatorLogic/CalculatorLogic.cs

```diff
@@ -0,0 +1,192 @@
+﻿// <copyright file="CalculatorLogic.cs" company="Anna Agafonova">
+// Copyright (c) Anna Agafonova. All rights reserved.
+// </copyright>
+
+/// <summary>
+/// Represents the core logic for a calculator application.
+/// </summary>
+public class CalculatorLogic : ViewModelBase
+{
+    private string displayedResult = string.Empty;
+    private string displayedOperations = string.Empty;
+
+    private double currentResult = 0;
+    private string currentOperator = string.Empty;
+    private string lastOperator = string.Empty;
```

Comment:
Все что вы заполнили -- дефолтные значение ваших типов и их можно не заполнять

```suggestion
    private string lastOperator;
```